torch/csrc/cuda/green_context.h

```diff
@@ -0,0 +1,213 @@
+#pragma once
+#include <ATen/cuda/CUDAEvent.h>
+#if defined(CUDA_VERSION) && !defined(USE_ROCM)
+#include <c10/cuda/driver_api.h>
+#include <cuda.h>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+#endif
+
+class GreenContext {
```

Comment:
Why there is a whole class implementation right in .h file? And why it's part of `libtorch_python` rather than `libtorch_cuda`?

Reply:
this is grandfathered in from Driss's original branch but we can change it